### ISSUE TITLE
[$1000]  LHN - Instead of user name, "chat report " is displayed

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#26592]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Launch app
2. Tap profile icon
3. Tap workspaces
4. Tap any Workspace
5. Tap Members
6. Tap invite
7. Enter any new non-existing account
8. Tap Next
9. Tap Invite
10. Navigate to LHN

## Expected Result:
The invited workspace user name should display in the first line of report in the LHN

## Actual Result:
"Chat report" is displayed in the first line of report in the LHN instead of the invited workspace user name

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [x] Android / native
- [ ] Android / Chrome
- [x] iOS / native
- [ ] iOS / Safari
- [ ] MacOS / Chrome / Safari
- [ ] MacOS / Desktop

**Version Number:** 1.3.62-1

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/4147dee5-c8ac-4358-b4f2-3298794b7349

**Expensify/Expensify Issue URL:**

**Issue reported by:** Applause - Internal Team

**Slack conversation:**

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01a5e746cb88e86963</li>
        <li>Upwork Job ID: 1698644415415676928</li>
        <li>Last Price Increase: 2023-09-28</li>
    </ul>
</details>

**Comment by conorpendergrast** [comment]
Oh yeah, weird: 

![image](https://github.com/Expensify/App/assets/5802477/379527a1-d5be-4cfe-8937-bedc93214783)

**Comment by im-adithya** [comment]
## Proposal

### Please re-state the problem that we are trying to solve in this issue.
"Chat Report" is displayed instead of username

### What is the root cause of that problem?
Because we send in `undefined` (second param, `reportName`) to `buildOptimisticChatReport` while creating the chat, see
https://github.com/Expensify/App/blob/f9f0ba58a4564470932473cd74fe9474bea04700/src/libs/actions/Policy.js#L290

### What changes do you think we should make in order to solve the problem?

We can send the email as the `reportName` instead of `undefined`

### What alternative solutions did you explore? (Optional)
< Thinking >

**Comment by conorpendergrast** [comment]
I'm on parental leave; re-assigning! This has been triaged and is waiting for proposals.

**Comment by mallenexpensify** [comment]
Bumped to $1000 to get more 👀 since it's been 10 days

**Comment by Julesssss** [comment]
@mallenexpensify @parasharrajat -- FYI there might be some discussion [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1694775748339609?thread_ts=1694699736.392239&cid=C01GTK53T8Q) in Slack, as I flagged it internally for more 👀 

If we're not able to verify, it might be worth assigning an engineer reviewer here, just so they can verify the above in case any API debugging is necessary.

**Comment by mallenexpensify** [comment]
@parasharrajat , please review @paultsimura 's proposal above, thx
https://github.com/Expensify/App/issues/26592#issuecomment-1721081902

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by paultsimura** [comment]
Sorry if that's something out of my realm. It's still reproducible on `1.3.72-9` (see the third chat).
Please make sure you're testing on iOS or Android @mvtglobally.

<img width="391" alt="image" src="https://github.com/Expensify/App/assets/12595293/a7d20c6d-3985-493b-84ed-674cd548f75d">

**Comment by mallenexpensify** [comment]
Reassigning C+ due to inactivity. 
@cubuspl42 can you please review the proposal above?
https://github.com/Expensify/App/issues/26592#issuecomment-1721081902

**Comment by cubuspl42** [comment]
@paultsimura 

Thank you for this extensive investigation! It looks really solid and it looks like we found the smoking gun here.

> TL;DR: Onyx processes the successData and API response in the wrong order on iOS.

> Please make sure you're testing on iOS or Android

Just to clarify, which platform(s) does this occur on?

While your root cause analysis is solid, the solution part looks not ready. For now, we're just throwing ideas, I don't think we have a proper _solution_ proposal yet.

**Comment by paultsimura** [comment]
@cubuspl42 I tested on iOS and Android – it happens on both platforms.
Unfortunately, even though I was able to track the issue to the root cause, I don't think I'm capable of suggesting a solution here. So let it be a starting point for somebody after me 🥲 
There was a slack discussion [here](https://expensify.slack.com/archives/C01GTK53T8Q/p1694775748339609?thread_ts=1694699736.392239&cid=C01GTK53T8Q), it might be helpful.

**Comment by cubuspl42** [comment]
@paultsimura I understand. This is a tough one, it seems. We can wait a bit for other proposals and maybe even raise the bounty.

Maybe it will be possible to assign a bonus for your investigation.

**Comment by Julesssss** [comment]
> Maybe it will be possible to assign a bonus for your investigation.

Yeah, we will definitely do this.

**Comment by paultsimura** [comment]
Thank you guys, that's very nice of you😌

**Comment by Julesssss** [comment]
I've shared this internally with one of our expert external agencies who hopefully will be able to verifiy and then fix this issue 🤞

**Comment by pradeepmdk** [comment]
the web also has this issue.  I think in issue in onyx.update functions because data is not presist in the database properly. 
https://github.com/Expensify/App/assets/16595846/1f0d7982-4b96-4969-9407-ebc60e2d0367


why we see initially the username means before presist the data in the database callback will be triggered. so that we can see that refresh.

**Comment by pradeepmdk** [comment]
https://github.com/Expensify/react-native-onyx/blob/82b5372aa6fef416408bbe71986029e18793e43a/lib/Onyx.js#L1367-L1408

here we are doing an async operation this will be a race condition

**Comment by cubuspl42** [comment]
@Julesssss So, is this issue classified as an agency issue now? Should we remove the "Help wanted" label? Or should I review the proposal and continue as usual?

**Comment by pradeepmdk** [comment]
@cubuspl42 @paultsimura  could you verify this issue again because now not able to reproduce it in staging.

because now recently updated react-native onyx fixed some performance issue as well.

https://github.com/Expensify/App/assets/16595846/f21dbd39-7892-4377-8675-9fb53f76eb25

**Comment by ospfranco** [comment]
I also just tested this on main and cannot reproduce it. Let me know if anyone is still having an issue.

**Comment by Julesssss** [comment]
I'm also unable to reproduce. but that doesn't prove that the out-of-order updates are no longer occuring, so I think we should do the following:
- Close this issue
- @paultsimura create a new bug for the out-of-order updates that you discovered, we can continue that investigation and pay our for your help separately from this issue

**Comment by paultsimura** [comment]
Thank you @Julesssss, I will report a bug and will tag you in the thread.

**Comment by Julesssss** [comment]
I'll keep this issue open until that issue is created, just so we don't forgot

**Comment by ospfranco** [comment]
Taking a look into other issues and it seems the issue might be triggered due to network latency. That might explain why everyone is getting results all over the place independently of the platform.

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Second week)

**Comment by mallenexpensify** [comment]
Closing cuz I think we were waiting for the below to be created, and it is!
- https://github.com/Expensify/App/issues/28737

**Comment by ospfranco** [comment]
My changes on https://github.com/Expensify/App/issues/28737 were reverted. The change caused issues in other places of the app. You should probably re-open this again.

**Comment by mallenexpensify** [comment]
Thanks @ospfranco

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (Third week)

**Comment by mallenexpensify** [comment]
Closing cuz it's not reproducible.
